pkg/asset/cluster/openstack/openstack.go

```diff
@@ -35,6 +45,51 @@ func PreTerraform() error {
 	} else if !errors.Is(err, os.ErrNotExist) {
 		return fmt.Errorf("unable to determine if clouds.yaml exists: %w", err)
 	}
+	return nil
```

Comment:
nit: This error checking logic is weird (IMO) and could be changed now that we're returning from here. Probably a follow-up PR.

Reply:
I agree it's weird. So much so, that I didn't want to touch it during the refactoring because I didn't want to inadvertently change its behaviour.